### ISSUE TITLE
Improve the implementation of `golem_api.event_bus.EventBus._emit`

### ISSUE THREAD

**Issue by johny-b** [golemfactory/golem-core-python#3]
Currently we wait for all callbacks for an event to finish before we start emitting next event.
This is obviously wrong - a single long callback just hangs the EventBus.

Our goal should be (imo, I'm not super-sure) to:
1. Ensure no event is ever lost (--> so we don't consider "just add a callback timeout" option)
2. Ensure event order is preserved **within particular callback**. So e.g. if I have a long callback A and a short callback B They can finish like this:
```
B, Event 1
B, Event 2
A, Event 1
A, Event 2
```
but never (X, Event 2) before (X, Event 1).

---> Solution is to have for each registered callback a separate task that executes the callbacks (--> probably also a separate event queue).

Edit: recent approach to the topic:
https://github.com/golemfactory/golem-core-python/commit/b7cd3bcf1445dc9aa4e0c03f59b1eb6c097ead40
